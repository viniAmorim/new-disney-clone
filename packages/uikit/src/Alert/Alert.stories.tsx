import React from 'react';

import { Story, Meta } from '@storybook/react';

import Alert, { AlertProps } from '~/Alert';

export default {
  title: 'Form/Alert',
  component: Alert,
} as Meta;

const Template: Story<AlertProps> = args => <Alert {...args} />;

export const Success = Template.bind({});
Success.args = {
  children:
    'Solicitação de senha enviada com sucesso. Em breve você receberá um e-mail com um link para criar uma nova senha.',
  appearance: 'success',
};
