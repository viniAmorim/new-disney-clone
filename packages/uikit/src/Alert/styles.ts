import styled, { css } from 'styled-components';

export type StyledAlertProps = {
  appearance?: 'success';
};

const appearanceModifier = {
  success: css`
    background: rgba(44, 211, 111);
    color: ${({ theme }) => theme.colors['gray-100']};
  `,
};

export const Alert = styled.div<StyledAlertProps>`
  transition: all 0.5s ease-out;

  border-radius: 4px;
  padding: 16px;
  position: fixed;
  top: 70px;
  right: 10px;

  ${({ appearance = 'success' }) => appearanceModifier[appearance]}
  ${({ theme }) => theme.typography['subheader-light']};
`;
