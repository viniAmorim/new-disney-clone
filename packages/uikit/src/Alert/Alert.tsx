import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface AlertProps
  extends BaseHTMLAttributes<HTMLDivElement>,
    Styled.StyledAlertProps {
  /**
   * Alert content.
   */
  children: ReactNode;
}

const Alert = ({ appearance = 'success', children, ...rest }: AlertProps) => {
  return (
    <Styled.Alert appearance={appearance} {...rest}>
      {children}
    </Styled.Alert>
  );
};

export default Alert;
