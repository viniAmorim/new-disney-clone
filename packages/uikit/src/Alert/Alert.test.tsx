import React from 'react';

import { render } from '~/test/utils';

import Alert from './Alert';

describe('Alert', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(<Alert data-testid="alert">Alert</Alert>);

    const alertElement = getByTestId('alert');

    expect(alertElement).toBeInTheDocument();
    expect(alertElement).toHaveTextContent(/^Alert$/);

    expect(alertElement).toMatchSnapshot();
  });
});
