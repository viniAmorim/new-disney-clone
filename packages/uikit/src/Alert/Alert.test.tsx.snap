// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly 1`] = `
<div
  class="sc-bdfBQB bGUESE"
  data-testid="alert"
>
  Alert
</div>
`;
